jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/support/opentsdb/OpenTSDBMessageFormatter.java

```diff
@@ -163,6 +172,10 @@ private void formatResultString(StringBuilder resultString, String metricName, l
 	}
 
 	public Iterable<String> formatResults(Iterable<Result> results) {
+		return formatResults(results, null);
+	}
+	public Iterable<String> formatResults(Iterable<Result> results, Server server) {
```

Comment:
Adding `Server` to the arguments of `formatResults` seems to make sense. But it should replace the previous `formatArguments`, not override it (and you should also modify all call sites). Passing `server` indirectly, via a class property is not thread safe and mixes what class state with call specific parameters.